apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/DisplayServiceAccountCard.tsx

```diff
@@ -0,0 +1,255 @@
+  Note,
+  Card,
+  Stack,
+  Paragraph,
+  FormControl,
+  Spinner,
+  Button,
+} from '@contentful/f36-components';
+import { ExternalLinkTrimmedIcon, CheckCircleIcon, ErrorCircleIcon } from '@contentful/f36-icons';
+import { useApi } from 'hooks/useApi';
+import { ServiceAccountKeyId, ServiceAccountKey } from 'types';
+
+interface Props {
+  serviceAccountKeyId: ServiceAccountKeyId;
+  serviceAccountKey: ServiceAccountKey;
+  onEditGoogleAccountDetails: React.MouseEventHandler<HTMLButtonElement>;
+  onAccountSummariesChange: Function;
+}
+
+export const GoogleErrorApiErrorTypes = {
+  AdminAPI: 'AdminApiError',
+  DataAPI: 'DataApiError',
+  Unknown: 'Unknown',
+};
+
+interface GAErrorApiType {
+  details: any;
+  errorType: string;
+  message: string;
+}
+
+const DisplayServiceAccountCard = (props: Props) => {
+  const {
+    serviceAccountKeyId,
+    serviceAccountKey,
+    onEditGoogleAccountDetails,
+    onAccountSummariesChange,
+  } = props;
+
+  const [isLoading, setIsLoading] = useState(true);
+  const [adminApiError, setAdminApiError] = useState<GAErrorApiType>();
+  const [dataApiError, setDataApiError] = useState<GAErrorApiType>();
+  const [unspecifiedAdminError, setUnspecifiedAdminError] = useState<GAErrorApiType>();
+  const [unspecifiedDataError, setUnspecifiedDataError] = useState<GAErrorApiType>();
+
+  // NOTE: Due to a bug installation parameters are not available at sdk.parameters.installation form the config screen
+  // location. Therefore we must pass down the values directly to the useApi hook. If the bug is fixed this won't be
+  // necessary
+  const api = useApi(serviceAccountKeyId, serviceAccountKey);
+
+  const verifyAdminApi = useCallback(async () => {
+    try {
+      setIsLoading(true);
+      const fetchedAccountSummaries = await api.listAccountSummaries();
+      if (fetchedAccountSummaries.res && !fetchedAccountSummaries.res.ok) {
+        setAdminApiError(fetchedAccountSummaries.errors);
+        return;
+      }
+
+      onAccountSummariesChange(fetchedAccountSummaries);
+      setAdminApiError(undefined);
+      setUnspecifiedAdminError(undefined)
+    } catch (e: any) {
+      setUnspecifiedAdminError({
+        details: null,
+        errorType: GoogleErrorApiErrorTypes.AdminAPI,
+        message: 'Unknown Google Admin API Error',
+      });
+    } finally {
+      setIsLoading(false);
+    }
+
+    return () => {
+      setAdminApiError(undefined);
+      setUnspecifiedAdminError(undefined)
+      setIsLoading(false);
+    }
+  
+  // It wants to add onAccountSummariesChange as a dependency but this will cause an infinite re-render
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [api]);
+
+  const verifyDataApi = useCallback(async () => {
+    try {
+      setIsLoading(true);
+      const pagedData = await api.getPageData();
+
+      // If we get a 400 it means auth check passes and api has been enabled
+      if (pagedData.res && !pagedData.res.ok && pagedData.res.status !== 400) {
```

Comment:
Makes sense. Can we be more specific in the error returned from GA though and maybe check the errorType instead of the status?